desc/objectives/linear_objectives.py

```diff
@@ -3238,3 +3238,193 @@ def compute(self, params, constants=None):
 
         """
         return params["Psi"]
+
+
+class FixCurveShift(_FixedObjective):
+    """Fixes Curve.shift attribute, which is redundant with other Curve params.
+
+    Parameters
+    ----------
+    curve : Curve
+        Curve that will be optimized to satisfy the Objective.
+    target : {float, ndarray}, optional
+        Target value(s) of the objective. Only used if bounds is None.
+        Must be broadcastable to Objective.dim_f.
+    bounds : tuple of {float, ndarray}, optional
+        Lower and upper bounds on the objective. Overrides target.
+        Both bounds must be broadcastable to to Objective.dim_f
+    weight : {float, ndarray}, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        Must be broadcastable to to Objective.dim_f
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True.
+    name : str, optional
+        Name of the objective function.
+
+    """
+
+    _target_arg = "shift"
+    _units = "(m)"
+    _print_value_fmt = "Fixed-shift error: {:10.3e} "
+
+    def __init__(
+        self,
+        curve,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        name="fixed-shift",
+    ):
+        self._target_from_user = setdefault(bounds, target)
+        super().__init__(
+            things=curve,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            name=name,
+        )
+
+    def build(self, use_jit=False, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        curve = self.things[0]
+        self._dim_f = curve.shift.size
+
+        self.target, self.bounds = self._parse_target_from_user(
+            self._target_from_user, curve.shift, None, np.arange(self._dim_f)
+        )
+
+        if self._normalize:
+            self._normalization = 1
```

Comment:
Really we should add normalization options for `Curve` to the `compute_scaling_factors` function, but for these objectives the normalization will never really matter.

Reply:
#863 can be easily done in a separate PR, this one is getting large